packages/core/src/mpi/get-patient-by-demo.ts

```diff
@@ -0,0 +1,60 @@
+import { Patient, PatientData } from "../domain/medical/patient";
+import { PatientLoader } from "../domain/medical/patient-loader";
+import {
+  jaroWinklerSimilarity,
+  matchingContactDetailsRule,
+  matchingPersonalIdentifiersRule,
+  matchPatients,
+} from "./match-patients";
+import { mergeWithFirstPatient } from "./merge-patients";
+import { normalizePatient } from "./normalize-patient";
+import { patientToPatientMPI } from "./shared";
+
+const SIMILARITY_THRESHOLD = 0.96;
+
+/**
+ * Retrieves a patient based on their demographic information. Utilizes functions
+ * from the MPI: normalization, finding(blocking), matching, merging
+ *
+ * @param cxId - The ID of the patient in the external system.
+ * @param demo - The demographic information of the patient.
+ * @returns The matched patient object if found, otherwise undefined.
+ */
+export const getPatientByDemo = async ({
+  cxId,
+  demo,
+  patientLoader,
+}: {
+  cxId: string;
+  demo: PatientData;
+  patientLoader: PatientLoader;
+}): Promise<Patient | undefined> => {
+  // Normalize the patient demographic data
+  const normalizedPatientDemo = normalizePatient(demo);
+  if (!normalizedPatientDemo) return undefined;
+
+  // Find patients based on the criteria of matching dob and genderAtBirth
+  const foundPatients = await patientLoader.findBySimilarity({
+    cxId,
+    data: {
+      dob: normalizedPatientDemo.dob,
+      genderAtBirth: normalizedPatientDemo.genderAtBirth,
+    },
+  });
+
+  // Convert patients to proper datatype
+  // Match the found patients with the normalized patient using the similarity function
+  const matchingPatients = matchPatients(
+    jaroWinklerSimilarity,
+    [matchingPersonalIdentifiersRule, matchingContactDetailsRule],
+    foundPatients.map(patientToPatientMPI),
+    normalizedPatientDemo,
+    SIMILARITY_THRESHOLD
+  );
+
+  // Merge the matching patients
+  const mpiPatient = mergeWithFirstPatient(matchingPatients, normalizedPatientDemo);
```

Comment:
Every time I read this `mergeWithFirstPatient(matchingPatients, normalizedPatientDemo)` I think it's merging `normalizedPatientDemo` with the first patient of `matchingPatients`, but it's not the case.

Can we rename this function to something that better represents what it does? Also, I don't like that we pass `normalizedPatientDemo` just so it can log and send it to Sentry, the code should not be built around log/reporting (e.g., every time we call this function we need the second parameter).

Reply:
changed to `useFirstMatchingPatient` and removed the `normalizedPatientDemo` from function call